### ISSUE TITLE
disable_monkey_patching url in spec helper template links to unavailable Relish documentation

### ISSUE THREAD

**Issue by tomczak-mateusz** [rspec/rspec-core#3031]
### Subject of the issue
disable_monkey_patching url in spec helper template links to unavailable Relish documentation

As I have read on page: https://rspec.info/documentation/ - RSpec documentation is being moved to rspec.info from RelishApp:

<img width="786" alt="Screenshot 2023-04-16 at 11 50 34" src="https://user-images.githubusercontent.com/99652987/232291187-fbd55232-175c-4b28-8f95-8a99583cbe52.png">

I have opened this issue as it might be confusing for the fresh RSpec users to hit the unavailable link.

### Your environment
* Ruby version: 3.2.2
* rspec-core version: 3.12.1

### Steps to reproduce
1. add rspec-rails to Rails application
2. run rails generate rspec:install
3. open generated spec_helper file
4. go to:
```
  # Limits the available syntax to the non-monkey patched syntax that is
  # recommended. For more details, see:
  # https://relishapp.com/rspec/rspec-core/docs/configuration/zero-monkey-patching-mode
  config.disable_monkey_patching!
```
5. open `For more details, see` link: https://relishapp.com/rspec/rspec-core/docs/configuration/zero-monkey-patching-mode
6. application error template should be shown with 503 error status
<img width="472" alt="Screenshot 2023-04-16 at 11 46 26" src="https://user-images.githubusercontent.com/99652987/232290991-c8f6a6cc-a2cc-4cad-b43e-8f265c68b255.png">

### Expected behavior
- spec_helper template should link to valid documentation link https://rspec.info/documentation/3.12/rspec-core/RSpec/Core/Configuration.html#disable_monkey_patching!-instance_method

### Actual behavior
- spec_helper template links to unavailable documentation link https://relishapp.com/rspec/rspec-core/docs/configuration/zero-monkey-patching-mode

**Comment by pirj** [comment]
@JonRowe If I replace `3.12` in https://rspec.info/documentation/3.12/rspec-core/RSpec/Core/Configuration.html#disable_monkey_patching!-instance_method to `latest`, it [redirects to the right page](https://rspec.info/documentation/latest/rspec-core/RSpec/Core/Configuration.html#disable_monkey_patching!-instance_method), but the `#disable_monkey_patching!-instance_method` anchor is lost.

**Comment by tomczak-mateusz** [comment]
@pirj hi, correct, it's the same case for me. 

btw I have tried to open Pull Request with updates for links from RelishApp to RSpec.info around the whole repository, but while doing so I'm getting 403 error:

```
➜  rspec-core git:(main) git checkout -b issue-3031
Switched to a new branch 'issue-3031'

➜  rspec-core git:(issue-3031) git push -u origin issue-3031
remote: Permission to rspec/rspec-core.git denied to tomczak-mateusz.
fatal: unable to access 'https://github.com/rspec/rspec-core.git/': The requested URL returned error: 403
```

any ideas what's wrong with this flow? :confused: it's not clear to me after following https://github.com/rspec/rspec-core#contributing

**Comment by pirj** [comment]
It's missing from the docs, but normally, you fork the repo, push the changes to your forked repo, and send a PR upstream.

**Comment by tomczak-mateusz** [comment]
@pirj thanks, gonna try the proposed path

@JonRowe  @pirj 
I would also like to suggest to update the GitHub repository 'About' url from : relishapp.com/rspec/rspec-core to https://rspec.info/documentation/latest/rspec-core/ as it might be currently confusing for the users that are not aware that Relish docs are down.

<img width="306" alt="Screenshot 2023-04-16 at 12 27 29" src="https://user-images.githubusercontent.com/99652987/232298143-fc616aef-fdf8-4742-9621-d2105105c8a0.png">

**Comment by pirj** [comment]
Nice catch. Other repos, too, and `rspec-rails` to https://rspec.info/documentation/latest/rspec-rails

**Comment by tomczak-mateusz** [comment]
@pirj @JonRowe  I have taken the opportunity to open the issues for 'About' url updates in each of the repos that mentions Relish for consistency & track ability:
 `rspec-core`: https://github.com/rspec/rspec-core/issues/3032
 `rspec-mocks`: https://github.com/rspec/rspec-mocks/issues/1543
 `rspec-expectations`: https://github.com/rspec/rspec-expectations/issues/1417

**Comment by tomczak-mateusz** [comment]
> @JonRowe If I replace `3.12` in https://rspec.info/documentation/3.12/rspec-core/RSpec/Core/Configuration.html#disable_monkey_patching!-instance_method to `latest`, it [redirects to the right page](https://rspec.info/documentation/latest/rspec-core/RSpec/Core/Configuration.html#disable_monkey_patching!-instance_method), but the `#disable_monkey_patching!-instance_method` anchor is lost.

@JonRowe if it's possible to fix rspec.info docs to work with latest in the url path and to not drop the anchor part like in the example above - I would be very happy to update all Relish related links in `rspec-core`, `rspec-rails`, `rspec-mocks`, and `rspec-expectations` to work with rspec.info. 

Please confirm if that's doable. :wink:

**Comment by pirj** [comment]
@tomczak-mateusz I guess this is handled [here](https://github.com/rspec/rspec.github.io/blob/1d47500e446df4ce74ca16b3f4c9fab6258df844/config.rb#L122) and [here](https://github.com/rspec/rspec.github.io/blob/1d47500e446df4ce74ca16b3f4c9fab6258df844/source/redirect-latest.html.slim#L2) if you feel adventurous enough to try fixing the missing anchor.

**Comment by JonRowe** [comment]
Fixed in #3035

**Comment by JonRowe** [comment]
Remove of the link in generated content was released in 3.12.2
